src/server/session.ts

```diff
@@ -1068,6 +1071,12 @@ export class Session<TMessage = string> implements EventSender {
     }
 
     private sendRequestCompletedEvent(requestId: number): void {
+        if (this.requestPerformanceData.has(requestId)) {
+            const diagnosticsDuration = arrayFrom(this.requestPerformanceData.get(requestId)!.entries()).map(([file, data]) => ({ file, ...data }));

```

Comment:
Two things

1. `arrayFrom` takes a mapper I believe
2. I might put the spread earlier to ensure stuff is never overwritten.

```suggestion
            const diagnosticsDuration = arrayFrom(this.requestPerformanceData.get(requestId)!.entries()), ([file, data]) => ({ ...data, file }));
```